cmd/gocritic/testdata/lenNil/negative_tests.go

```diff
@@ -0,0 +1,25 @@
+package checker_test
+
+func g1() bool {
+	var m map[int]int
+	if m == nil && len(m) == 0 {
+		return false
+	}
+	var ch chan int
+	return ch != nil && len(ch)+10 == 100
+}
+
+func g2() {
+	var ch chan int
+	if ch == nil {
+		return
+	}
+	if len(ch) == 0 {
+		return
+	}
+
+	switch ch != nil && len(ch)+10 == 100 {
+	case true:
+	default:
+	}
```

Comment:
What about `x != nil || len(x) == 0`?
I don't see `Tok == token.LAND` check inside the implementation.

Reply:
Also, I don't see checks for `len` argument.
This code may result in a false positive: `x != nil && len(y) == 0`.